extensions/guacamole-auth-sso/modules/guacamole-auth-sso-saml/src/main/java/org/apache/guacamole/auth/saml/conf/ConfigurationService.java

```diff
@@ -371,7 +371,13 @@ public int getAuthenticationTimeout() throws GuacamoleException {
      *     If the X.509 certificate cannot be parsed.
      */
     public File getCertificateFile() throws GuacamoleException {
-        return environment.getProperty(SAML_X509_CERT_PATH);
+    	File certificate = null;
+    	try {
+			certificate = environment.getProperty(SAML_X509_CERT_PATH).getCanonicalFile();
```

Comment:
This will throw a `NullPointerException` if `getProperty()` returns `null` (if the property is not present). Same for the other call to `getCanonicalFile()`.